zeus/device/soc/__init__.py

```diff
@@ -17,17 +17,28 @@ def get_soc() -> SoC:
     The function returns a SoC management object that aims to abstract underlying SoC monitoring
     functionalities.
 
-    Currently, no management object has been implemented for any SoC architecture, so calling this
-    function will raise a `ZeusSoCInitError` error; implementations for SoC devices are expected
-    to be added in the near future.
+    Currently supported SoC devices:
+        - Apple Silicon
+
+    If no SoC monitor object can be initialized, a `ZeusSoCInitError` exception will be raised.
     """
     global _soc
     if _soc is not None:
         return _soc
 
-    # SoCs in the future can be incorporated via `elif` blocks.
-    else:
-        # Placeholder to avoid linting error (remove once _soc can be assigned a real value).
-        _soc = None
+    # --- Apple Silicon ---
+    try:
+        # The below import is done here instead of at top of file because the
+        # Apple Silicon module contains optional dependencies that will cause
+        # import failures if not installed on the host device.
+        from zeus.device.soc.apple import AppleSilicon, ZeusAppleInitError
+
+        _soc = AppleSilicon()
+    except (ImportError, ZeusSoCInitError):
+        pass
```

Comment:
There should be a `is_apple_silicon_available` function exposed by `apple.py`. Inside `apple.py`, it'll import `zeus_apple_silicon` from within a `try`-`except` clause and if an exception is raised, it'll just set `zeus_apple_silicon` to a mock module class. See:

https://github.com/ml-energy/zeus/blob/0d5087f8b3f3d2dedc92ffee066af579513f403b/zeus/device/gpu/amd.py#L11-L33